app/controllers/account/tasks_controller.rb

```diff
@@ -1,56 +1,68 @@
 class Account::TasksController < Account::AccountController
-  before_action :set_project
-  before_action :set_task, only: [:show, :edit, :update, :destroy]
-
   def show
+    @project = parent
+    @task = @project.tasks.find(params[:id])
   end
 
   def new
-    @task = @project.tasks.new
+    @project = parent
+    @task = @project.tasks.build
   end
 
   def create
-    @task = @project.tasks.new(tasks_params)
+    @task = collection.build(tasks_params)
 
     if @task.save
-      redirect_to account_project_path(@project)
+      redirect_to account_workspace_project_path(parent.workspace_id, parent)
     else
       render :new
     end
   end
 
   def edit
+    @project = parent
+    @task = @project.tasks.find(params[:id])
   end
 
   def update
+    @task = resource
+
     if @task.update(tasks_params)
-      redirect_to account_project_task_path(@project, @task)
+      redirect_to account_project_task_path(parent.id, @task)
     else
       render "edit"
     end
   end
 
   def destroy
-    @task.destroy
-    redirect_to account_project_path(@project)
+    resource.destroy
+    redirect_to account_workspace_project_path(parent.workspace_id, parent.id)
   end
 
   def move
-    Task.find(params[:task_id]).update_attribute(:row_order_position, params[:move])
-    redirect_to account_project_path(@project)
+    resource.update_attributes(task_movement_params)
+    redirect_to account_workspace_project_path(parent.workspace_id, parent.id)
   end
 
   private
 
-  def set_project
-    @project = current_user.projects.find(params[:project_id])
+  def parent
+    Project.find(params[:project_id])
```

Comment:
so I can put any ID in URL and do something with project and tasks? where is currrent_user?